### PR TITLE
feat(base-repo): included the base repository in chargejs

### PR DESCRIPTION
**Pull request by dmytro-kulchytskyi**
https://charge.atlassian.net/browse/WALL-1062

### CODE REVIEW

File: src/infra/BaseRepository.js [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+          transaction: this._getTransaction()
+        }
+      )
+      if (!result) {
+        throw this._getNotFoundError()
+      }
+      return this.mapper.toEntity(result.toJSON())
+    } catch (error) {
+      switch (error.name) {
+        case 'SequelizeValidationError':
```
Comment (NFhbar) [inline src/infra/BaseRepository.js:231]: put these strings in the `errors` file.

File: src/infra/BaseRepository.js [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+  }
+
+  _getTransaction () {
+    return this.transactionProvider.getSequelizeTransaction()
+  }
+
+  _filterPatchFields (updateFields) {
+    if (!this.patchAllowedFields) {
+      // Error for developers only, can only occur if the patchAllowedFields configuration is incorrect
+      const error = new Error('patchAllowedFields was not configured in init function')
```
Comment (NFhbar) [inline src/infra/BaseRepository.js:318]: put these `strings` in the errors file.

File: src/infra/BaseRepository.js [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+
+  _validationErrorMessage () {
+    const modelMessages = this.logmsg.info[this.modelName] || {}
+    const name = this._getCapitalizedModelName()
+    return modelMessages.validationError || `Invalid ${name} field data`
+  }
+
+  _getCacheDisabledError () {
+    // Error for developers only, can only occur if the disableCache configuration is incorrect
+    const error = new Error('disableCache option was set to true, please provide provide disableCache: false option to use caching.')
```
Comment (NFhbar) [inline src/infra/BaseRepository.js:376]: put these `strings` in the error file.

File: src/infra/redis/RedisRepository.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+  }
+
+  _validateFilter (filter) {
+    const normalizedFilter = this._normalizeFilter(filter)
+    const keys = Object.keys(normalizedFilter).sort()
+    const valid = !!this.cacheKeys
+      .find(key => isEqual(key, keys))
+    if (!valid) {
+      // Error for developers only, can only occur if the cacheKeys configuration is incorrect
+      const error = new Error(`Invalid filter. Configure cache keys to support provided filter fields: "${JSON.stringify(keys)}"`)
```
Comment (NFhbar) [inline src/infra/redis/RedisRepository.js:102]: put in `error` file.

File: src/infra/redis/RedisStorage.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+   * @memberof module:repository.RedisRepository
+   */
+  async saveObject (key, object) {
+    try {
+      const redisObject = JSON.stringify(object)
+      const reply = await this.redisClient.setAsync(key, redisObject)
+      return reply
+    } catch (error) {
+      this.logger.error({
+        event: 'create',
```
Comment (NFhbar) [inline src/infra/redis/RedisStorage.js:38]: put these in the `strings` in error file.

**Review by NFhbar — CHANGES_REQUESTED** [review]
Awesome!
Just a few comments on moving all `strings` to the `error` and/or `constants` file.
Also, is this a `major` change?

**Comment by dmytro-kulchytskyi** [comment]
> Awesome!
> Just a few comments on moving all `strings` to the `error` and/or `constants` file.
> Also, is this a `major` change?

@NFhbar There are a lot of changes, but I haven't broken any existing functionality, so I think it's a minor

**Review by NFhbar — APPROVED** [review]

**Review by amanelis — APPROVED** [review]
epic ☄️

**Review by supaflyENJOY — APPROVED** [review]

**Review by valera-rozuvan — APPROVED** [review]

**Pull request merged**